garena_parser.py

```diff
@@ -0,0 +1,244 @@
+match_data = match_data_and_timeline[0]
+match_timeline = match_data_and_timeline[1]
+
+patch = match_data["gameVersion"]
+patch = ".".join(patch.split(".", 2)[:2])
+ddragon_patch = patch + ".1"
+
+runes = requests.get(
+    f"https://raw.communitydragon.org/{patch}/plugins/rcp-be-lol-game-data/global/default/v1/perks.json")
+runes = runes.json()
+champions = requests.get(f"https://ddragon.leagueoflegends.com/cdn/{ddragon_patch}/data/en_US/champion.json")
+champions = champions.json()["data"]
+items = requests.get(f"https://ddragon.leagueoflegends.com/cdn/{ddragon_patch}/data/en_US/item.json")
+items = items.json()["data"]
+spells = requests.get(f"https://ddragon.leagueoflegends.com/cdn/{ddragon_patch}/data/en_US/summoner.json")
+spells = spells.json()["data"]
+
+champion_ids = {}
+spell_ids = {}
+rune_ids = {
+    8000: "Precision",
+    8100: "Domination",
+    8200: "Sorcery",
+    8300: "Inspiration",
+    8400: "Resolve"
+}
+item_ids = {}
+
+for key, champion in champions.items():
+    champion_ids[int(champion["key"])] = champion["name"]
+
+for key, spell in spells.items():
+    spell_ids[int(spell["key"])] = spell["name"]
+
+for rune in runes:
+    rune_ids[rune["id"]] = rune["name"]
+
+for key, item in items.items():
+    item_ids[int(key)] = item["name"]
+
+item_ids[0] = ""
+
+game_length = int(match_data["gameDuration"])
+game_length_pretty = '{}:{}'.format(str(math.floor(game_length / 60)), str(int(round(game_length % 60, 0))).zfill(2))
+
+game_start_timestamp = int(match_data["gameCreation"] / 1000)
+game_start = datetime.utcfromtimestamp(game_start_timestamp)
+kst = timezone("Asia/Seoul")
+kst_object = game_start.astimezone(kst)
+start_date = kst_object.strftime("%Y-%m-%d")
+start_time = kst_object.strftime("%H:%M")
+
+matchschedule_info = site.cargo_client.query(
+    tables="MatchScheduleGame=MSG, MatchSchedule=MS, Tournaments=T",
+    fields="MSG.Blue, MSG.Red, MSG.Winner, MS.DateTime_UTC, MS.DST, MS.OverviewPage, T.StandardName",
+    where=f"MSG.RiotPlatformGameId = '{rpgid.upper()}'",
+    join_on="MSG.MatchId = MS.MatchId, MS.OverviewPage=T.OverviewPage"
+)
+if len(matchschedule_info) > 1 or not matchschedule_info:
+    print("More than one game found in data!")
+    exit()
+
+matchschedule_info = matchschedule_info[0]
+teams = {"team1": matchschedule_info["Blue"], "team2": matchschedule_info["Red"]}
+winner_data = str(matchschedule_info["Winner"])
+dst = matchschedule_info["DST"]
+tournament = matchschedule_info["StandardName"]
+
+for team in match_data["teams"]:
+    if team["win"] == True or team["win"] == "Win":
+        winner_team_id = int(team["teamId"])
+
+if winner_team_id == 100:
+    winner = "1"
+elif winner_team_id == 200:
+    winner = "2"
+else:
+    print("No winner team could be found!")
+    exit()
+
+if winner != winner_data:
+    print("Winner team does not match!")
+    exit()
+
+# teamstats = {"team1": {}, "team2": {}}
+
+participantstats = []
+participantids_dict = {
```

Comment:
similar to EXTRA_RUNE_IDS, this should be a constant at the start

also don't put the type in the variable name, call it like PARTICIPANT_IDS_NAMES